compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -669,12 +669,6 @@ pub(super) fn collect_return_position_impl_trait_in_trait_tys<'tcx>(
         )
         .fold_with(&mut collector);
 
-    debug_assert_ne!(
```

Comment:
I don't think you should remove this assertion. The only way this assertion is not correct is when the code has other errors.